pkg/resource/stack/deployment.go

```diff
@@ -209,6 +210,42 @@ func SerializeDeployment(ctx context.Context, snap *deploy.Snapshot, showSecrets
 	}, nil
 }
 
+// SerializeOptions controls how a deployment is serialized to JSON.
+type SerializeOptions struct {
+	// ShowSecrets indicates that secrets should be shown in the serialized deployment.
+	ShowSecrets bool
+	// Pretty indicates that the serialized deployment should be indented and formatted for display to users.
+	Pretty bool
```

Comment:
Do we even need Pretty? Looks like all the uses of encoding.JSON are in test files? Does it hurt to just change them to json.Marshal?

Reply:
See my other reply. We need it for `pulumi stack export --show-secrets`.